### PR TITLE
Retry bootstrap if it fails

### PR DESCRIPTION
**Pull request by chromakode**
This will retry the bootstrap 3 times, with increasing delay (3s, 5s, then 10s). This doesn't improve the UX of the bootstrap finally failing, though -- should we ticket that separately, or should I follow this up with another PR for indicating failure to the user?

:eyeglasses: @keybase/react-hackers

### CODE REVIEW

**Comment by chrisnojima** [comment]
I thought during planning we discussed not using the approach. Instead of
it fails we were going to reset the connection

**Comment by chromakode** [comment]
Ah, I had recalled us discussing retries. The ticket didn't have much information -- we should have updated it with the details of that conversation. I'll whip up a version that resets the connection instead of retrying the bootstrap.

**Comment by chromakode** [comment]
@chrisnojima Ok, tweaked to reset the engine instead of just running bootstrap again. Does this match what we had intended? My memory is a bit hazy 🙃

**Comment by chromakode** [comment]
Yes I did test it this way:

```
chromakode@atavist:~/dev/keybase/synced/client$ git stash show -p
diff --git a/shared/actions/config/index.js b/shared/actions/config/index.js
index 0c0b4b5..4e23fa5 100644
--- a/shared/actions/config/index.js
+++ b/shared/actions/config/index.js
@@ -16,6 +16,9 @@ import {configGetConfigRpc, configGetExtendedStatusRpc, configGetCurrentStatusRp
 function getConfig (): AsyncAction {
   return (dispatch, getState) => {
     return new Promise((resolve, reject) => {
+      if (getState().config.bootstrapTriesRemaining > 0) {
+        reject()
+      }
       configGetConfigRpc({
         callback: (error, config) => {
           if (error) {
```

Worked on desktop but it [doesn't look like we're able to reset the client on mobile yet](https://github.com/keybase/client/blob/679b4065ed2d6dac1bcec2d03f2a57d8fe0bb014/shared/engine/index.js#L221). Thoughts on what to do for mobile?

**Comment by chrisnojima** [comment]
Well let's ticket

File: shared/actions/config/index.js [hunk 1]
```diff
@@ -120,15 +120,16 @@ export function bootstrap (): AsyncAction {
...
       engine().listenOnConnect('bootstrap', () => {
-        console.log('Bootstrapping')
+        console.log('[bootstrap] bootstrapping on connect')
         dispatch(bootstrap())
       })
     } else if (getState().dev.reloading) {
       // Let's still register the listeners
       dispatch(_registerListeners())
     } else {
+      console.log('[bootstrap] performing bootstrap...')
```
Comment (chrisnojima) [inline shared/actions/config/index.js]: we don't still want this right? we see the actions being logged already

Comment (chromakode) [inline shared/actions/config/index.js]: I wanted this to differentiate whether we hit the `bootstrapSetup` branch vs. actually bootstrappging. Do you think it is superfluous?

Comment (chrisnojima) [inline shared/actions/config/index.js]: your call

**Comment by chromakode** [comment]
@chrisnojima Ok this now calls [`reset`](https://github.com/keybase/node-framed-msgpack-rpc/blob/master/src/transport.iced#L146) on the framed-msgpack transport, triggering a reconnect as we discussed. I've tested it w/ simulating getConfig failures and the reconnect loop seems to work.

**Comment by chrisnojima** [comment]
👍

**Comment by chromakode** [comment]
@chrisnojima needed to make a small fix to flow here: https://github.com/keybase/client/pull/4157/commits/e8c6c0ca481cbf49baca206514583c42c12924ef

**Comment by chrisnojima** [comment]
👍

**Pull request merged**